### PR TITLE
Mise en place certificate pinning

### PR DESCRIPTION
**Pull request by yostyle**
## Content

Add certificate pinning feature only on Gplay release build. 

## Motivation and context

https://github.com/tchapgouv/tchap-product/issues/373

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: app/src/fdroid/res/xml/network_security_config.xml
Comment (NicolasBuquet) [inline app/src/fdroid/res/xml/network_security_config.xml:1]: Why does F-Droid version need to allow clear traffic on local network ? VPN need on NEO devices ?

**Review by NicolasBuquet — APPROVED** [review]
LGTM!

File: app/src/fdroid/res/xml/network_security_config.xml
Comment (yostyle) [reply app/src/fdroid/res/xml/network_security_config.xml:1]: Clear text traffic can be used only for specific hosts during development :
- localhost
- 127.0.0.1
- etc ...

It's the default configuration provided by Element X for all variants (Fdroid and Gplay). I move it for Fdroid only.

**Comment by NicolasBuquet** [comment]
@yostyle Attention à la configuration du Certificate Pinning sur la couche réseau du Rust SDK : https://github.com/tchapgouv/tchap-x-ios/issues/36

**Pull request merged**